sagemaker_processing/spark_distributed_data_processing/sagemaker-spark-processing.ipynb

```diff
@@ -0,0 +1,562 @@
+    "                                outputCol=\"features\")\n",
+    "    \n",
+    "    # The pipeline comprises of the steps added above\n",
+    "    pipeline = Pipeline(stages=[sex_indexer, sex_encoder, assembler])\n",
+    "    \n",
+    "    # This step trains the feature transformers\n",
+    "    model = pipeline.fit(total_df)\n",
+    "    \n",
+    "    # This step transforms the dataset with information obtained from the previous fit\n",
+    "    transformed_total_df = model.transform(total_df)\n",
+    "    \n",
+    "    # Split the overall dataset into 80-20 training and validation\n",
+    "    (train_df, validation_df) = transformed_total_df.randomSplit([0.8, 0.2])\n",
+    "    \n",
+    "    # Convert the train dataframe to RDD to save in CSV format and upload to S3\n",
+    "    train_rdd = train_df.rdd.map(lambda x: (x.rings, x.features))\n",
+    "    train_lines = train_rdd.map(csv_line)\n",
+    "    train_lines.saveAsTextFile('s3://' + os.path.join(args['s3_output_bucket'], args['s3_output_key_prefix'], 'train'))\n",
+    "    \n",
+    "    # Convert the validation dataframe to RDD to save in CSV format and upload to S3\n",
+    "    validation_rdd = validation_df.rdd.map(lambda x: (x.rings, x.features))\n",
+    "    validation_lines = validation_rdd.map(csv_line)\n",
+    "    validation_lines.saveAsTextFile('s3://' + os.path.join(args['s3_output_bucket'], args['s3_output_key_prefix'], 'validation'))\n",
+    "\n",
+    "\n",
+    "if __name__ == \"__main__\":\n",
+    "    main()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Run the SageMaker Processing Job\n",
+    "\n",
+    "Next, you'll use the `PySparkProcessor` class to define a Spark job and run it using SageMaker Processing. A few things to note in the definition of the `PySparkProcessor`:\n",
+    "\n",
+    "* This is a multi-node job with 2x m5.xlarge instances (which is specified via the `instance_count` and `instance_type` parameters)\n",
+    "* Spark framework version 2.4 is specified via the `framework_version` parameter\n",
+    "* The PySpark script defined above is passed via via the `submit_app` parameter\n",
+    "* Arguments are being passed to the PySpark script (such as the s3 input and output locations) via the `arguments` parameter\n",
+    "* Spark event logs will be offloaded to the s3 location specified in `spark_event_logs_s3_uri` and can be used to view the Spark UI while the job is in progress or after it completes\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sagemaker.spark.processing import PySparkProcessor\n",
+    "\n",
+    "# Upload the raw input dataset to a unique S3 location\n",
+    "timestamp_prefix = strftime(\"%Y-%m-%d-%H-%M-%S\", gmtime())\n",
+    "prefix = 'sagemaker/spark-preprocess-demo/' + timestamp_prefix\n",
```

Comment:
Suggested edit:

Favor `.format` or `f-strings` and double-quotes as convention, here and throughout:

```suggestion
    "prefix = f\"sagemaker/spark-preprocess-demo/{timestamp_prefix}\"\n",
```